lib/AST/ASTScopeCreation.cpp

```diff
@@ -210,6 +210,8 @@ void ASTScope::
 
 void ASTScope::expandFunctionBody(AbstractFunctionDecl *AFD) {
   auto *const SF = AFD->getParentSourceFile();
+  if (!SF)
```

Comment:
For whatever reason, it looks like the accessor is unable to find a source file in the parent decl contexts. I'm going to investigate further.